sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/Utility.java

```diff
@@ -187,4 +190,21 @@ Pair readBytes(int cnt) {
             return this;
         }
     }
+
+    /**
+     * Mapping a {@link ErrorResponseException} to {@link HttpResponseException} if exist. Otherwise, return
+     * original {@link Throwable}.
+     *
+     * @param throwable A {@link Throwable}.
+     * @return A {@link HttpResponseException} or the original throwable type.
+     */
+    public static Throwable mapToHttpResponseExceptionIfExist(Throwable throwable) {
```

Comment:
@srnagar Does this seem like something we would want the autorest to be doing for the client ?

Reply:
Yes, I think it is better to define in swagger or make it configurable on swagger readme. Search has the same problem.

Reply:
AutoRest will always provide the most specific error response. It's for the convenience layer to determine if it should be converted to a more generic exception type.